service/history/queues/util.go

```diff
@@ -56,3 +58,20 @@ func IsTaskAcked(
 
 	return true
 }
+
+// IsTaskDone is similar to IsTaskAcked but it performs some additional checks
+func IsTaskDone(shard shard.Context, category tasks.Category, workflowKey definition.WorkflowKey, taskID int64) bool {
+	if taskID == 0 {
+		return true
+	}
+	persistenceQueueState, ok := shard.GetQueueState(category)
+	if !ok {
+		// Use global ack level visibility queue ack level because cluster level is not updated.
+		visibilityQueueAckLevel := shard.GetQueueAckLevel(category).TaskID
+		return taskID <= visibilityQueueAckLevel
+	}
+
+	fakeTask := tasks.NewFakeTask(workflowKey, category, tasks.DefaultFireTime)
```

Comment:
We can't use fake task here, task type information is necessary for `IsTaskAcked` to determine if a task is still pending or not.

Reply:
Task fire time information is also required, so defaultFireTime probably can't work.

Reply:
Fixed